### DISCUSSION TITLE
error 400 issues using docker installation

### DISCUSSION THREAD

**Discussion by Runningman-work** [semaphoreui/semaphore#2193] [Q&A]
I am using semaphore 2.10.11 and installed it with following docker compose file:

> services:
>   db:
>     container_name: semaphore-db
>     restart: unless-stopped
>     env_file: stack.env
>     image: postgres:16
>     hostname: postgres
>     volumes: 
>       - /storage/semaphore/postgres:/var/lib/postgresql/data
>     environment:
>       - POSTGRES_USER=${POSTGRES_USER}
>       - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
>       - POSTGRES_DB=${POSTGRES_DB}
>   app:
>     container_name: semaphore-app
>     restart: unless-stopped
>     env_file: stack.env
>     ports:
>       - 3000:3000
>     image: semaphoreui/semaphore:latest
>     environment:
>       - SEMAPHORE_DB_USER=${POSTGRES_USER}
>       - SEMAPHORE_DB_PASS=${POSTGRES_PASSWORD}
>       - SEMAPHORE_DB_HOST=postgres
>       - SEMAPHORE_DB_PORT=5432
>       - SEMAPHORE_DB_DIALECT=postgres
>       - SEMAPHORE_DB=${POSTGRES_DB}
>       - SEMAPHORE_PLAYBOOK_PATH=${SEMAPHORE_PLAYBOOK_PATH}
>       - SEMAPHORE_ADMIN_PASSWORD=${SEMAPHORE_ADMIN_PASSWORD}
>       - SEMAPHORE_ADMIN_NAME=${SEMAPHORE_ADMIN_NAME}
>       - SEMAPHORE_ADMIN_EMAIL=${SEMAPHORE_ADMIN_EMAIL}
>       - SEMAPHORE_ADMIN=${SEMAPHORE_ADMIN}
>       - SEMAPHORE_ACCESS_KEY_ENCRYPTION=${SEMAPHORE_ACCESS_KEY_ENCRYPTION}
>       - TZ=Europe/Berlin
>     depends_on:
>       - db

When I try to create a login key in the key store, I receive an error 400. It happens with every key as Login or ssh type, only "none" type works.

Whats wrong?

**Comment by fiftin** [comment]
Please try latest Beta

**Reply by Runningman-work** [reply]
> Please try latest Beta

I did a fresh install of v2.10.16-beta, same error 400.

**Comment by Runningman-work** [comment]
Found the problem:

There was an error in the  docker logs of the app:

`time="2024-07-23T08:40:08+02:00" level=error msg="illegal base64 data at input byte 1"`

The encryption key must be created like this:
`head -c32 /dev/urandom | base64`

[see also](https://github.com/semaphoreui/semaphore/issues/1758)
